controller.go

```diff
@@ -96,62 +49,7 @@ func CreateVolume(
 		return nil, err
 	}
 
-	// check to see if there is a csi error
-	if cerr := res.GetError(); cerr != nil {
-		if err := cerr.GetCreateVolumeError(); err != nil {
-			return nil, fmt.Errorf(
-				"error: CreateVolume failed: %d: %s",
-				err.GetErrorCode(),
-				err.GetErrorDescription())
-		}
-		if err := cerr.GetGeneralError(); err != nil {
-			return nil, fmt.Errorf(
-				"error: CreateVolume failed: %d: %s",
-				err.GetErrorCode(),
-				err.GetErrorDescription())
-		}
-		return nil, errors.New(cerr.String())
-	}
-
-	result := res.GetResult()
```

Comment:
Ah, now that I read the interceptors, I see why this can go away. It looks like it can go away in lots more places. Is that in-progress?